src/test/java/gameframework/game/GameUniverseDefaultImplTest.java

```diff
@@ -0,0 +1,96 @@
+package gameframework.game;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import gameframework.game.mocks.GameDataMock;
+import gameframework.game.mocks.GameEntityMock;
+import gameframework.game.mocks.GameMovableEntityMock;
+import gameframework.game.mocks.GameMoveBlockerEntityMock;
+import gameframework.game.mocks.GameOverlappableEntityMock;
+import gameframework.motion.blocking.MoveBlockerChecker;
+import gameframework.motion.overlapping.mocks.OverlapProcessorDefaultImplMock;
+
+public class GameUniverseDefaultImplTest {
+
+	protected GameUniverseDefaultImpl GameUniverseDefaultImpl;
+	
+	@Before
+	public void init() {
+		GameUniverseDefaultImpl = new GameUniverseDefaultImpl();
+		GameUniverseDefaultImpl.setGameData(new GameDataMock());
+	}
+	
+	public void addAndRemoveEntity(GameEntity entity) {
+		GameUniverseDefaultImpl.addGameEntity(entity);
+		assertTrue(GameUniverseDefaultImpl.getGameEntities().contains(entity));
+		
+		GameUniverseDefaultImpl.removeGameEntity(entity);
+		assertFalse(GameUniverseDefaultImpl.getGameEntities().contains(entity));
+	}
+	
+	@Test
+	public void addAndRemoveGameEntity() {
+		GameEntityMock GameEntity = new GameEntityMock();
+		this.addAndRemoveEntity(GameEntity);
+	}
+	
+	@Test
+	public void addAndRemoveOverlappableEntity() {
+		GameOverlappableEntityMock OverlappableEntity = new GameOverlappableEntityMock();
+		
+		// OverlapProcessorDefaultImplMock is instanced by GameConfigurationMock, included in GameDataMock
+		OverlapProcessorDefaultImplMock OverlapProcessor = (OverlapProcessorDefaultImplMock) GameUniverseDefaultImpl.getOverlapProcessor();
+
+		GameUniverseDefaultImpl.addGameEntity(OverlappableEntity);
+		assertTrue(GameUniverseDefaultImpl.getGameEntities().contains(OverlappableEntity));
+		assertTrue(OverlapProcessor.test_isAdding(OverlappableEntity));
```

Comment:
I need to check the behavior of the method. I check if the addition was well done.